src/MongoDB/WriteConcern.c

```diff
@@ -298,7 +298,7 @@ static PHP_METHOD(WriteConcern, isDefault)
 	RETURN_BOOL(mongoc_write_concern_is_default(intern->write_concern));
 } /* }}} */
 
-static HashTable* php_phongo_write_concern_get_properties_hash(phongo_compat_object_handler_type* object, bool is_debug, bool is_bson) /* {{{ */
+static HashTable* php_phongo_writeconcern_get_properties_hash(phongo_compat_object_handler_type* object, bool is_temp, bool is_bson, bool is_serialize) /* {{{ */
```

Comment:
Before introducing `is_serialize`, `WriteConcern::serialize()` would enter the `! is_bson` code path and conditionally encode `wtimeout` based on the system architecture. That actually seemed like a mistake, since it could lead to a 64-bit system producing serialized output that cannot be processed on a 32-bit system (e.g. if `wtimeout` was `INT32_MAX + 1`).

This adds a new code path to ignore the system architecture and instead encode according to the range of the value itself to maximum compatibility. The `init_from_hash` function already supports both formats, so this isn't the same issue that came up with CursorId earlier.

Reply:
For what it's worth, looking at our platform stats, I think we'll stop considering 32-bit systems soon.